cala-server/src/graphql/schema.rs

```diff
@@ -111,6 +112,105 @@ impl<E: MutationExtensionMarker> CoreMutation<E> {
         Ok(journal.into_values().into())
     }
 
+    async fn tx_template_create(
+        &self,
+        ctx: &Context<'_>,
+        input: TxTemplateCreateInput,
+    ) -> Result<TxTemplateCreatePayload> {
+        let app = ctx.data_unchecked::<CalaApp>();
+        let id = if let Some(id) = input.id {
+            id.into()
+        } else {
+            cala_ledger::TxTemplateId::new()
+        };
+        let mut new_tx_input_builder = cala_ledger::tx_template::NewTxInput::builder();
+        let TxTemplateTxInput {
+            effective,
+            journal_id,
+            correlation_id,
+            external_id,
+            description,
+            metadata,
+        } = input.tx_input;
+        new_tx_input_builder
+            .effective(effective.as_ref().to_string())
```

Comment:
Why this as_ref().to_string() repeating pattern. Can probably improve this by accepting a trait